src/lib/storage/table.cpp

```diff
@@ -87,5 +88,32 @@ void Table::_append_new_chunk() {
   _chunks.push_back(std::move(new_chunk));
 }
 
-void Table::compress_chunk(ChunkID chunk_id) { throw std::runtime_error("Implement Table::compress_chunk"); }
+void Table::compress_chunk(ChunkID chunk_id) {
+  const auto& uncompressed_chunk = get_chunk(chunk_id);
+  Chunk compressed_chunk = Chunk();
+
+  std::vector<std::future<std::shared_ptr<BaseSegment>>> compressed_segment_futures;
+
+  const auto col_count = column_count();
+  for (ColumnID column_id = ColumnID{0}; column_id < col_count; ++column_id) {
+    const auto uncompressed_segment = uncompressed_chunk.get_segment(column_id);
+    std::promise<std::shared_ptr<BaseSegment>> promise;
+    compressed_segment_futures.push_back(promise.get_future());
+    auto thread = std::thread(compress_segment, std::move(promise), column_type(column_id), uncompressed_segment);
```

Comment:
We could do
```c++
std::thread thread(compress_segment, std::move(promise), column_type(column_id), uncompressed_segment);
```
I think this would be more consistent.

Reply:
Done